packages/core/src/common/types.ts

```diff
@@ -130,3 +132,25 @@ export namespace ArrayUtils {
 export function unreachable(_never: never, message: string = 'unhandled case'): never {
     throw new Error(message);
 }
+
+/**
+ * @internal
+ *
+ * Potentially empty Theia contribution types can cause TypeScript to complain
+ * if extra methods are defined. This type prevents the issue.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type EmptyMeansAny<T> = {} extends T ? T & { [key in string | symbol]: any } : T;
+
+/**
+ * @internal
+ */
+export type SymbolIdentifier<T> = symbol & interfaces.Abstract<EmptyMeansAny<T>>;
+
+/**
+ * Use this to create "typed symbols" for Inversify typings to understand our bindings.
+ */
+export function serviceIdentifier<T>(name: string): SymbolIdentifier<T> {
```

Comment:
Changing service identifiers from symbols to their own types is one change that could be done in a separate PR to make this one smaller. It affects a lot of files that need to be reviewed.

Reply:
I'll see to extract this once I get back to a functional state with this PR.

Right now I only used `serviceIdentifier` where I happened to work, but I could try to make a PR that replaces everything instead.